pkg/controllers/manila/manila.go

```diff
@@ -123,6 +147,66 @@ func (c *ManilaController) sync(ctx context.Context, syncCtx factory.SyncContext
 	return c.setEnabledCondition(ctx)
 }
 
+func (c *ManilaController) getFsGroupPolicy(ctx context.Context) storagev1.FSGroupPolicy {
+
+	// Set the fsGroupPolicy param to the default value
+	fsGroupPolicy := "None"
+
+	// Try to get the current value defined for the operator
+	if c.isFSGroupPolicyAllowed(os.Getenv("CSI_FSGROUP_POLICY")) {
+		fsGroupPolicy = os.Getenv("CSI_FSGROUP_POLICY")
+	}
+
+	return storagev1.FSGroupPolicy(string(fsGroupPolicy))
+}
+
+func (c *ManilaController) syncCSIDriver(ctx context.Context, spec storagev1.CSIDriverSpec) error {
+	klog.V(4).Infof("Starting CSI driver config refresh")
+	defer klog.V(4).Infof("CSI driver config refresh finished")
+
+	var errs []error
+
+	crd, err := c.kubeClient.StorageV1().CSIDrivers().Get(ctx, "manila.csi.openstack.org", metav1.GetOptions{})
+
+	if err != nil {
+		// no CSIDriverObject found, read from the asset and apply
+		stream, e := assets.ReadFile("csidriver.yaml")
+		if e != nil {
+			klog.V(4).Infof("Error loading the CSIDriver resource")
+		}
+		crd = resourceread.ReadCSIDriverV1OrDie(stream)
+	}
+
+	crd.Spec = spec
```

Comment:
If `Get()` above returns error (e.g. CSIDriver does not exist), `crd` is nil at this point.

IMO what you want in the whole `syncCSIDriver()` is:

```go
stream, err := assets.ReadFile("csidriver.yaml")
if err != nil {
	panic("Error loading the CSIDriver resource", err)
}

cr = resourceread.ReadCSIDriverV1OrDie(stream)
cr.Spec.FSGroupPolicy = getFSGroupPolicy()

_, _, err = resourceapply.ApplyCSIDriver(ctx, c.kubeClient.StorageV1(), c.eventRecorder, cr)
return err
```

Reply:
ack, so basically we're not reading the existing object anymore: we're just loading the asset and customizing the spec as defined in the manifest.
This is simple and less error prone: the last update defines this kind of behavior.